lib/tools/adb-commands.js

```diff
@@ -53,14 +53,18 @@ methods.initZipAlign = async function () {
 /**
  * Retrieve the API level of the device under test.
  *
- * @return {string} The API level as integer number, for example 21 for
+ * @return {number} The API level as integer number, for example 21 for
  *                  Android Lollipop. The result of this method is cached, so all the further
  * calls return the same value as the first one.
  */
 methods.getApiLevel = async function () {
-  if (!this._apiLevel) {
+  if (!_.isNumber(this._apiLevel)) {
     try {
-      this._apiLevel = await this.getDeviceProperty('ro.build.version.sdk');
+      const strOutput = await this.getDeviceProperty('ro.build.version.sdk');
+      this._apiLevel = parseInt(strOutput.trim(), 10);
+      if (isNaN(this._apiLevel)) {
```

Comment:
If `this._apiLevel` is `NaN`, if this method is called again `NaN` will be returned, since `_.isNumber(NaN) === true`. Either after this check `this._apiLevel` should be set to `null`/`undefined`, or the first check should include a `isNaN` check, or, since API level should always be an integer, the check could be `_.isInteger(this._apiLevel)`.

Reply:
_.isNumber(), which returns true for NotANumber... Something is wrong with this world

Reply:
NotANumber is definitely a number, the "not" kind!